src/Http/BaseApplication.php

```diff
@@ -110,6 +118,10 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
      */
     protected function getDispatcher()
     {
-        return new ActionDispatcher(null, null, DispatcherFactory::filters());
+        if ($this->dispatcher === null) {
```

Comment:
Shouldn't this be done in the constructor? Otherwise the Application is 'invalid' until this method is called to create a side-effect.